kmir/src/tests/integration/test_prove.py

```diff
@@ -38,13 +39,45 @@ def test_handwritten(
 
     # When
     try:
-        prove(
+        (passed, failed) = prove(
             kmir,
             spec_file=spec_file,
             save_directory=use_directory,
             smt_timeout=300,
             smt_retry_limit=10,
         )
+
+        # Check if passed proofs are in show
+        if len(passed) != 0:
+            for proof in passed:
+                if proof.id in SHOW_TESTS:
```

Comment:
Please create a seperate testing file for `show_proof` and testing the `show_proof` function instead of `prove` function.

Reply:
This is how I thought to implement it at first too, but I see why kontrol went the way of including the check in their [prove tests](https://github.com/runtimeverification/kontrol/blob/0c089b7b5f073412b3919832d7cc2478da2eccd3/src/tests/integration/test_foundry_prove.py#L132-L181). Show depends on a proof to be run to use for show, so if we separate it out I think it would end up with duplication of running proofs. This structure we just store what `proof.id` we have a saved show output for, and those proofs get checked for show as they are already being run.